src/stream/stream_backend.rs

```diff
@@ -220,6 +220,75 @@ fn create_udp_stream(
     )));
 }
 
+fn create_rtsp_stream(
+    video_and_stream_information: &VideoAndStreamInformation,
+) -> Result<StreamType, SimpleError> {
+    let encode = video_and_stream_information
+        .stream_information
+        .configuration
+        .encode
+        .clone();
+    let endpoints = &video_and_stream_information.stream_information.endpoints;
+    let configuration = &video_and_stream_information
+        .stream_information
+        .configuration;
+    let video_source = &video_and_stream_information.video_source;
+
+    let video_format = match video_source {
+        VideoSourceType::Local(local_device) => {
+            if VideoEncodeType::H264 == encode {
+                format!(
+                    concat!(
+                        "v4l2src device={device}",
+                        " ! video/x-h264,width={width},height={height},framerate={interval_denominator}/{interval_numerator}",
+                    ),
+                    device = &local_device.device_path,
+                    width = configuration.width,
+                    height = configuration.height,
+                    interval_denominator = configuration.frame_interval.denominator,
+                    interval_numerator = configuration.frame_interval.numerator,
+                )
+            } else {
```

Comment:
the if part is already a return, you don't need the else here.

Reply:
To explain my logic here, the `format!` code return an string to the inner scope, but the return is used to the outer scope.
The code is near to the following snippet:
```
fn potato_fun() -> &'static str {
    if 12 * 15 > 150 {
        "Bigger"
    } else {
        "Smaller"
    }
}
// rework
fn potato_fun() -> &'static str {
    if 12 * 15 > 150 {
        "Bigger"
    }

    "Smaller"
}
```

The problem is that the second case is invalid, it appears to be related to the let if pattern: https://doc.rust-lang.org/book/ch03-05-control-flow.html#using-if-in-a-let-statement